### PR TITLE
Cleanup obsolete NodeInfo methods

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Remove the last obsolete methods from scheduler's NodeInfo.

**Which issue(s) this PR fixes**:
Part of #89528

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
/uncc

I don't have a ton of context here, but definitely support these changes conceptually :)

**Comment by gavinfish** [comment]
This can also help #74352, thanks @ahg-g

**Comment by ahg-g** [comment]
@wojtek-t can you help us with this refactoring?

**Comment by wojtek-t** [comment]
/assign

Yes - I will take a look

File: pkg/kubelet/cm/container_manager.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
 	"k8s.io/kubernetes/pkg/kubelet/pluginmanager/cache"
 	"k8s.io/kubernetes/pkg/kubelet/status"
-	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
```
Comment (wojtek-t) [inline pkg/kubelet/cm/container_manager.go:34]: framework is super generic. especially if used in the packages like kubelet (and it's about scheduler).

I would suggest doing "schedulerframework" instead.
[So effectively comparing to the original version: s/schedulernodeinfo/schedulerframework/]

Comment (ahg-g) [reply pkg/kubelet/cm/container_manager.go]: done for all occurrences outside pkg/scheduler

**Pull request merged**